src/views/InteractiveMap/InteractiveMap.vue

```diff
@@ -457,59 +457,95 @@ export default Vue.extend({
           hasSimulationError: false
         }
       });
-      axios
-        .post(`${settings.apis.model}/simulate`, {
-          model_id: model.id,
-          method: card.method,
-          operations: operations,
-          objective_id: card.objective.reaction
-            ? card.objective.reaction.id
-            : model.default_biomass_reaction,
-          objective_direction: card.objective.maximize ? "max" : "min"
-        })
-        .then(response => {
-          const wildtypeFluxdistribution = response.data.flux_distribution;
-          // Calculate new bounds based on manipulation scores calculated by DiffFVA
-          // (reference flux * score) / production growth
-          const editedBounds = card.manipulations
-            .map(function(manipulation) {
-              const newBound = Math.round(
-                (wildtypeFluxdistribution[manipulation.id] *
-                  manipulation.value) /
-                  card.productionGrowthRate
-              );
+      // Apply the values from diffFVA results as bounds for simulation
+      const editedBounds = card.manipulations.map(function(manipulation) {
+        // do we need to filter out values that are zero or is this controlled on the backend?
+        const model_reaction = model.model_serialized.reactions.find(
+          rxn => rxn.id === manipulation.id
+        );
+        const newBound = manipulation.value;
+        switch (manipulation.direction) {
+          case "up":
+            if (newBound > 0) {
               return {
                 operation: "modify",
                 type: "reaction",
                 id: manipulation.id,
                 data: {
                   lower_bound: newBound,
-                  upper_bound: newBound
+                  upper_bound: model_reaction.upper_bound
```

Comment:
In some cases, I'm getting `model_reaction is undefined` error here:

![image](https://user-images.githubusercontent.com/64745/61120821-13791080-a49e-11e9-9e9f-3e98f304aba8.png)

It happens consistently when I visualize a DiffFVA design for one of my jobs, I'd be happy to try to help debug it.

Reply:
I think it is important that you have to have diffFVA results generated with the newest backend. Did you make sure to test it with recent results?

Reply:
Oh but it could also be the added reactions I suppose?

Reply:
Could you confirm that this happened with newer jobs?

Reply:
No it doesn't seem to happen with newer jobs.

Reply:
I was able to reproduce this with a newer job after all. It happens when one of the manipulations is for one of the heterologous reactions. You're trying to find that reaction in the wild type model, which of course isn't there. I'll start work on a fix in a separate PR to be merged to this one.

Reply:
Fixed in https://github.com/DD-DeCaF/caffeine-vue/pull/117